airqo-platform/AirQo-api
# Extract external data Microservice

# Data management microservice for extracting meta data from external providers

**_WHAT DOES THIS PR DO?_**
 Contains utilities for extracting data (metadata) from third party providers and exposing them as api endpoints for utilization in other data management services
**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
   -  https://airqoteam.atlassian.net/browse/DP-76?atlOrigin=eyJpIjoiNTZhYWIxODgxYzk3NGE2NTkxMDk2ODk2MTE2MmNhY2MiLCJwIjoiaiJ9
**_WHAT IS THE LINK TO THE PR BRANCH_**
[Link to PR Branch](https://github.com/airqo-platform/AirQo-api/compare/extract_external_data_ms)
**_HOW DO I TEST OUT THIS PR?_**
- Git fetch the branch and checkit out locally
- cd directory to AirQo-api\src\data-mgt\python\extract-external-data
-  Include the .env & private key file
  -- check platformapi/keys/data-mgt/python/extract-external-data on the shared drive
- build and run the staging docker image 
  -- docker build -t <container_name> -f Dockerfile.stage .
  -- docker run -it  -p 4000:4001<container_name>:latest
**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
[Check out the endpoints on documentation ](https://app.gitbook.com/@airqo/s/airqo-platform-api/extract-data-from-external-sources)
 Substitute the base url (https://staging-platform.airq.net) with http://127.0.0.1:4000
**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
N/A
**_ARE THERE ANY RELATED PRs?_**
N/A
